src/main/kotlin/com/openlattice/transporter/types/TransporterDatastore.kt

```diff
@@ -0,0 +1,199 @@
+            """.trimMargin()
+                    .split("\n")
+
+            queries.forEach {
+                logger.info("about to run {}", it)
+            }
+            queries.forEach { sql ->
+                logger.info("running {}", sql)
+                st.execute(sql)
+            }
+            conn.commit()
+        }
+    }
+
+    fun importTablesFromForeignSchema(
+            remoteSchema: String,
+            remoteTables: Set<String>,
+            localSchema: String,
+            usingFdwName: String
+    ): String {
+        val tablesClause = if ( remoteTables.isEmpty() ){
+            ""
+        } else {
+            "LIMIT TO (${remoteTables.joinToString(",")})"
+        }
+        return "import foreign schema $remoteSchema $tablesClause from server $usingFdwName INTO $localSchema;"
+    }
+
+    private fun ensureSearchPath( hds: HikariDataSource ): String {
+        logger.info("checking search path for current user")
+        hds.connection.use { conn ->
+            val st = conn.createStatement()
+            st.executeQuery( "show search_path" ).use {
+                it.next()
+                val searchPath = it.getString(1)
+                logger.info(searchPath)
+                if ( searchPath == null ) {
+                    logger.error("bad search path: {}", searchPath)
+                    return ""
+                }
+                return searchPath
+            }
+        }
+    }
+
+    private fun initializeFDW(rhizomeConfig: PostgresConfiguration) {
+        createFdwBetweenDatabases(
+                hds,
+                rhizomeConfig.hikariConfiguration.getProperty("username"),
+                rhizomeConfig.hikariConfiguration.getProperty("password"),
+                rhizomeConfig.hikariConfiguration.getProperty("jdbcUrl"),
+                assemblerConfiguration.server.getProperty("username"),
+                enterpriseFdwSchema,
+                enterpriseFdwName
+        )
+
+        hds.connection.use { conn ->
+            conn.createStatement().use { stmt ->
+                stmt.executeQuery("select count(*) from information_schema.foreign_tables where foreign_table_schema = 'ol'").use { rs ->
+                    if (rs.next() && rs.getInt(1) > 0) {
+                        // don't bother if it's already there
+                        logger.info("schema already imported, not re-importing")
+                    } else {
+                        stmt.executeUpdate(
+                                importTablesFromForeignSchema(
+                                        "public",
+                                        setOf(),
+                                        enterpriseFdwSchema,
+                                        enterpriseFdwName
+                                )
+                        )
+                    }
+                }
+            }
+        }
+
+        hds.close()
+        hds = exConnMan.connect("transporter")
```

Comment:
I feel like db names should also be constantified -- I've relied on finding usage for things like this several times

Reply:
👍